gammapy/image/maps.py

```diff
@@ -416,13 +417,18 @@ def lookup_max(self, region=None):
 
     def solid_angle(self):
         """
-        Solid angle image
+        Solid angle image (2-dim `astropy.units.Quantity` in `sr`).
         """
         coordinates = self.coordinates(mode='edges')
-        xsky = coordinates.data.lon
-        ysky = coordinates.data.lat
-        omega = np.abs(np.diff(xsky, axis=1)[1:, :] * np.diff(ysky, axis=0)[:, 1:])
-        return omega.to('sr')
+        lon = coordinates.data.lon.radian
+        lat = coordinates.data.lat.radian
+
+        dx = angular_separation(lon[:, :-1], lat[:, :-1],
+                                    lon[:, 1:], lat[:, :-1])
+
+        dy = angular_separation(lon[:-1, :], lat[:-1, :],
+                                lon[1:, :], lat[1:, :])
+        return Quantity(dx[1:, :] * dy[:, 1:], "sr")
```

Comment:
No, currently slicing is done twice, which is incorrect. When computing the separation `dx` in x - direction, the second axes `lat` is already sliced with `lat[:, :-1]` and the same is done for `dy` in the y - direction. For the two step slicing one would have to keep the second axis untouched, like so: 
```
dx = angular_separation(lon[:, :-1], lat,  lon[:, 1:], lat)
dy = angular_separation(lon, lat[:-1, :],  lon, lat[1:, :])
```
Your implementation seems correct to me.